### PR TITLE
doc/doxygen: Enable MathJax

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

We definitely need LaTeX formulas in our doc! Why? Because we can!

### Testing procedure

Add a LaTeX formula, e.g. `\f( \mathbf{E} \cdot \mathrm{d} \mathbf{S} ={ \frac{1}{\varepsilon_{0}}}\iiint_{\Omega}\rho \, \mathrm{d} V \f)`, somewhere in a Doxygen comment. The generated HTML doc should display a proper formula, when opened in a web browser with JavaScript support enabled.

### Issues/PRs references

None

### CODE REVIEW

**Comment by maribu** [comment]
Note: Mainly commit https://github.com/RIOT-OS/RIOT/pull/19481/commits/78a44e432bd492da32e6e320a49fbf49454b711e which updates the Doxyfile via `doxygen -u` contributes to the large diff. Actually enabling MathJax support is a one-line change.

**Comment by maribu** [comment]
```
   Command output:
  
  	warning: ignoring unsupported tag 'CREATE_SUBDIRS_LEVEL' at line 93, file -
  	warning: ignoring unsupported tag 'SHOW_HEADERFILE' at line 623, file -
  	warning: ignoring unsupported tag 'WARN_IF_INCOMPLETE_DOC' at line 841, file -
  	warning: ignoring unsupported tag 'WARN_IF_UNDOC_ENUM_VAL' at line 859, file -
  	warning: ignoring unsupported tag 'WARN_LINE_FORMAT' at line 889, file -
  	warning: ignoring unsupported tag 'INPUT_FILE_ENCODING' at line 960, file -
  	warning: ignoring unsupported tag 'FORTRAN_COMMENT_AFTER' at line 1134, file -
  	warning: ignoring unsupported tag 'HTML_COLORSTYLE' at line 1370, file -
  	warning: ignoring unsupported tag 'DOCSET_FEEDURL' at line 1471, file -
  	warning: ignoring unsupported tag 'FULL_SIDEBAR' at line 1682, file -
  	warning: ignoring unsupported tag 'OBFUSCATE_EMAILS' at line 1713, file -
  	warning: ignoring unsupported tag 'MATHJAX_VERSION' at line 1761, file -
  	warning: ignoring unsupported tag 'DOT_COMMON_ATTR' at line 2483, file -
  	warning: ignoring unsupported tag 'DOT_EDGE_ATTR' at line 2492, file -
  	warning: ignoring unsupported tag 'DOT_NODE_ATTR' at line 2500, file -
  	warning: ignoring unsupported tag 'DIR_GRAPH_MAX_DEPTH' at line 2650, file -
```

I guess I shouldn't have updated the Doxyfile then :) I'll drop the commit.

**Comment by maribu** [comment]
This seems to just be broken with the Doxygen version in the CI :/

File: core/lib/include/clist.h [hunk 1]
```diff
@@ -19,23 +19,23 @@
...
- * clist_find_before()  | O(n)    | find node return node pointing to node
- * clist_remove()       | O(n)    | remove and return node
- * clist_sort()         | O(NlogN)| sort list (stable)
- * clist_count()        | O(n)    | count the number of elements in a list
- * clist_is_empty()     | O(1)    | returns true if the list contains no elements
- * clist_exactly_one()  | O(1)    | returns true if the list contains one element
- * clist_more_than_one()| O(1)    | returns true if the list contains more than one element
+ * operation            | runtime                                                   | description
+ * ---------------------|-----------------------------------------------------------|-----------------------------------------------------------
+ * clist_lpush()        | \f(\mathcal{O}\left(1\right)\f)                           | insert as head (leftmost node)
```
Comment (kfessel) [inline core/lib/include/clist.h:24]: this mixes  something readable in the plain realy unreadable

Comment (kfessel) [inline core/lib/include/clist.h:24]: this mixes  something readable in the plain realy unreadable

File: doc/doxygen/riot.doxyfile [hunk 2]
```diff
@@ -1478,7 +1478,7 @@ FORMULA_TRANSPARENT    = YES
 # The default value is: NO.
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
-USE_MATHJAX            = NO
+USE_MATHJAX            = YES
```
Comment (kfessel) [inline doc/doxygen/riot.doxyfile:1481]:
```suggestion
USE_MATHJAX            = NO
```

alle uses shown above decrease readabilty by a lot ( most people do not use the rended output of the doxgen

**Review by kfessel — CHANGES_REQUESTED** [review]
decrease readabilty

consider this: strong vote against

U missed April 1st by a lot

**Pull request closed without merging**